### ISSUE TITLE
Add bloom post-processing effect

### ISSUE THREAD

**Issue by cdata** [GoogleWebComponents/model-viewer#210]

**Comment by jcguarinpenaranda** [comment]
Hello, I've seen this has been opened for a while. May I help with something?

**Comment by elalish** [comment]
@jcguarinpenaranda Absolutely! Contributions are very welcome, and this should be reasonably doable given that three.js already supports it: https://threejs.org/examples/webgl_postprocessing_unreal_bloom.html

We'll want it to be off by default since it has render cost. What kind of parameterization are you considering?

**Comment by ti-hardikshah** [comment]
Is there any example of how can we use bloom with the model viewer?

**Comment by elalish** [comment]
No, but you can fork us and add Bloom in our three.js code using their examples. If you'd like to contribute it back when you get it working, that would be much appreciated.

**Comment by AhrenFullStop** [comment]
nobody up for the task? I might have a stab

**Comment by elalish** [comment]
@AhrenFullStop By all means! Happy to review. I should really expose a rendering performance metric so we can track how expensive these these features are. I don't have a good sense for it at the moment. As long as it  defaults to off it should be fine though.

**Comment by braedenf** [comment]
This would be an amazing feature to integrate. I was thinking of trying something like this example - https://p3d.in/jCj24

**Comment by Beilinson** [comment]
hi, will this be implemented any time soon?

**Comment by elalish** [comment]
Not on my roadmap, but PRs welcome!

**Comment by Beilinson** [comment]
> Not on my roadmap, but PRs welcome!

It should be possible by simply replacing the [`threeRender.render(scene, scene.camera)` line in the `Renderer`](https://github.com/google/model-viewer/blob/61435b8b2aaf160b92317b8d27e00cf8e86eefb3/packages/model-viewer/src/three-components/Renderer.ts#L494) with `composer.render()` from this [example](https://threejs.org/docs/#manual/en/introduction/How-to-use-post-processing), right? Or is there something else I'm missing in the rendering stack?
